src/snowcli/cli/registry/commands.py

```diff
@@ -14,10 +18,116 @@
 
 
 @app.command("token")
-@with_output
 @global_options_with_connection
-def token(environment: str = ConnectionOption, **options):
+@with_output
+def token(**options):
     """
     Gets the token from environment to use for authenticating with the registry.
     """
-    return ObjectResult(get_token(environment))
+    return ObjectResult(RegistryManager().get_token())
+
+
+@app.command("list-images")
+@with_output
+@global_options_with_connection
+def list_images(
+    repo_name: str = typer.Option(
+        ...,
+        "--repository_name",
+        "-r",
+        help="Name of the image repository as seen in `show image repositories`",
+    ),
+    **options,
+) -> MessageResult:
+    registry_manager = RegistryManager()
+    database = registry_manager.get_database()
+    schema = registry_manager.get_schema()
+    url = registry_manager.get_repository_url(repo_name)
+    api_url = registry_manager.get_repository_api_url(url)
+    bearer_login = registry_manager.login_to_registry(api_url)
+
+    repos = []
+    query: Optional[str] = f"{api_url}/_catalog?n=10"
+
+    while query:
+        # Make paginated catalog requests
+        response = requests.get(
+            query, headers={"Authorization": f"Bearer {bearer_login}"}
+        )
+
+        if response.status_code != 200:
+            return MessageResult(f"Call to the registry failed {response.text}")
+
+        data = json.loads(response.text)
+        if "repositories" in data:
+            repos.extend(data["repositories"])
+
+        if "Link" in response.headers:
+            # There are more results
+            query = f"{api_url}/_catalog?n=10&last={repos[-1]}"
+        else:
+            query = None
+
+    message = "Images in this repository:\n\n"
+    for repo in repos:
+        prefix = f"{database}/{schema}/{repo_name}/"
+        repo = repo.replace("baserepo/", prefix)
+        message = f"{message}{repo}\n"
+
+    return MessageResult(message)
```

Comment:
```suggestion
    images = []
    for repo in repos:
        prefix = f"{database}/{schema}/{repo_name}/"
        repo = repo.replace("baserepo/", prefix)
        images.append({"image": repo})

    return CollectionResult(images)
```
In this way we will get images structured as tables + if `--format=json` is used we will be able to pip the result to `jq`.